include/zephyr/drivers/dma/sf32lb.h

```diff
@@ -0,0 +1,184 @@
+	{                                                                                          \
+		.dev = DEVICE_DT_GET(DT_DMAS_CTLR_BY_NAME(node_id, name)),                         \
+		.channel = DT_DMAS_CELL_BY_NAME(node_id, name, channel),                           \
+		.request = DT_DMAS_CELL_BY_NAME(node_id, name, request),                           \
+		.config = DT_DMAS_CELL_BY_NAME(node_id, name, config),                             \
+	}
+
+/**
+ * @brief Initialize a `sf32lb_dma_dt_spec` structure from a DT instance.
+ *
+ * @param index DT instance index
+ */
+#define SF32LB_DMA_DT_INST_SPEC_GET(index) SF32LB_DMA_DT_SPEC_GET(DT_DRV_INST(index))
+
+/**
+ * @brief Initialize a `sf32lb_dma_dt_spec` structure from a DT instance.
+ *
+ * @param index DT instance index
+ * @param name DMA name
+ */
+#define SF32LB_DMA_DT_INST_SPEC_GET_BY_NAME(index, name)                                           \
+	SF32LB_DMA_DT_SPEC_GET_BY_NAME(DT_DRV_INST(index), name)
+
+/**
+ * @brief Check if the DMA controller is ready
+ *
+ * @param spec SF32LB DMA DT spec
+ *
+ * @return true If the DMA controller is ready.
+ * @return false If the DMA controller is not ready.
+ */
+static inline bool sf32lb_dma_is_ready_dt(const struct sf32lb_dma_dt_spec *spec)
+{
+	return device_is_ready(spec->dev);
+}
+
+/**
+ * @brief Initialize a DMA configuration structure from a DT spec.
+ *
+ * This function sets the following fields in the dma_config structure:
+ * - dma_slot
+ * - channel_priority
+ *
+ * @param spec SF32LB DMA DT spec
+ * @param[out] config DMA configuration
+ */
+static inline void sf32lb_dma_config_init_dt(const struct sf32lb_dma_dt_spec *spec,
+					     struct dma_config *config)
+{
+	config->dma_slot = spec->request;
+	config->channel_priority = FIELD_GET(SF32LB_DMA_PL_MSK, spec->config);
+}
+
+/**
+ * @brief Configure a DMA channel from a DT spec.
+ *
+ * @param spec SF32LB DMA DT spec
+ * @param config DMA configuration
+ *
+ * @retval 0 If successful.
+ * @retval -ENOTSUP If the configuration is not supported.
+ * @retval -errno Other negative errno code failure (see dma_config()).
+ */
+static inline int sf32lb_dma_config_dt(const struct sf32lb_dma_dt_spec *spec,
+				       struct dma_config *config)
+{
+	return dma_config(spec->dev, spec->channel, config);
+}
+
+/**
+ * @brief Start a DMA transfer from a DT spec.
+ *
+ * @param spec SF32LB DMA DT spec
+ *
+ * @retval 0 If successful.
+ * @retval -errno Negative errno code failure (see dma_start()).
+ */
+static inline int sf32lb_dma_start_dt(const struct sf32lb_dma_dt_spec *spec)
+{
+	return dma_start(spec->dev, spec->channel);
+}
+
+/**
+ * @brief Stop a DMA transfer from a DT spec.
+ *
+ * @param spec SF32LB DMA DT spec
+ *
+ * @retval 0 If successful.
+ * @retval -errno Negative errno code failure (see dma_stop()).
+ */
+static inline int sf32lb_dma_stop_dt(const struct sf32lb_dma_dt_spec *spec)
```

Comment:
There's basically no point in these public APIs, the device and channel being members of another peripheral device struct is highly common, as is using multiple channels (tx/rx pair).

Is there a reason to add these here?

Reply:
I mean, whatever honestly, but its odd to make this specific to sf32lb I guess. I don't expect this pattern to expand much further but if it does expect a refactor to common code at some point. Not a blocker from me.

Reply:
This sort of breaks the promise of a generic DMA interface when using devicetree, which would be the point if SoCs have IP blocks that are found elsewhere. We should modify the API so it can use dt-spec contents. However, it won't be easy, as DMA cells tend to be controller-specific (see Linux, for instance).